File: src/main/java/pl/training/workshop/m3/s10_isp/start/ScheduleBoard.java
package pl.training.workshop.m3.s10_isp.start;

import java.time.LocalTime;

/** Klient: tablica seansów. Używa scheduleScreening, cancelScreening i screenings. */
public final class ScheduleBoard {
    private final CinemaAdminService backOffice;

    public ScheduleBoard(CinemaAdminService backOffice) {
        this.backOffice = backOffice;
    }

    public void plan(String title, LocalTime start) {
        backOffice.scheduleScreening(title, start);
    }

    public void cancel(String title) {
        backOffice.cancelScreening(title);
    }

    public String board() {
        return String.join(", ", backOffice.screenings());
    }
}

File: typescript/src/workshop/m3/s10_isp/start/ScheduleBoard.ts
import type { LocalTime } from '../../../shared/time.js';
import type { CinemaAdminService } from './CinemaAdminService.js';

/** Klient: tablica seansów. Używa scheduleScreening, cancelScreening i screenings. */
export class ScheduleBoard {
  constructor(private readonly backOffice: CinemaAdminService) {}

  plan(title: string, start: LocalTime): void {
    this.backOffice.scheduleScreening(title, start);
  }

  cancel(title: string): void {
    this.backOffice.cancelScreening(title);
  }

  board(): string {
    return this.backOffice.screenings().join(', ');
  }
}
